daemon/src/model/cfd.rs

```diff
@@ -944,8 +956,19 @@ impl Cfd {
         Ok(self.event(cfd_event))
     }
 
-    pub fn handle_refund_timelock_expired(self) -> Event {
-        todo!()
+    pub fn handle_refund_timelock_expired(self) -> Result<Event, RefundTimelockExpiryError> {
+        use RefundTimelockExpiryError::*;
+
+        if self.is_final() {
+            return Err(AlreadyFinal);
+        }
```

Comment:
Is this actually an error or should we just emit `None`?

Reply:
I think the error handling in `cfd_actors` highlights that this is not actually an error.

Reply:
Strictly speaking it's not an error, but I found it more ergonomic both here and in the call site to treat all the "non-success" cases as part of the same `enum`.